runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -523,44 +632,188 @@ MM_SchedulingDelegate::calculateEstimatedGlobalBytesToScan() const
+	 *  We add additional headroom so that we don't ever exhaust that free space
+	 */
+	uintptr_t freeTenure = OMR_MAX((uintptr_t)(_estimatedFreeTenure * freeTenureHeadroom), 1);
+
+	if (0 == _totalGMPWorkTimeUs) {
+		/* We haven't seen a GMP yet, so _estimatedFreeTenure will still be 0, which is not accurate. Use another estimate for free tenure until a GMP happens*/
+		intptr_t freeTenureFromPGCInfo = (intptr_t)currentHeapSize - currentIdealEdenSize - _liveSetBytesAfterPartialCollect - (intptr_t)_averageSurvivorSetRegionCount;
+		freeTenure = freeTenureFromPGCInfo > 0 ? freeTenureFromPGCInfo : 1;
+	}
+
+	Assert_MM_true(freeTenure != 0);
+
+	/* Determine how far we can increase or decrease eden from where eden currently stands. */
+	intptr_t minEdenChange = (intptr_t)currentIdealEdenSize * -1; 
+	intptr_t maxEdenChange = (intptr_t)freeTenure;
+
+	/* How many samples we want to test between minEdenChange and maxEdenChange? */
+	uintptr_t numberOfSamples = 100;
+
+	/* 
+	 * Initially, we suggest the current eden size as the best size - until proven there is a better size.
+	 * The "better" size, will have a better blend of gc overhead (% of time gc is active relative to mutator), 
+	 * and more satisfactory pgc pause time (below target pgc pause is the goal).
+	 */
+	intptr_t recommendedEdenChange = 0;
+	double currentCpuEdenOverhead = predictCpuOverheadForEdenSize(env, currentIdealEdenSize, recommendedEdenChange, freeTenure, avgPgcIntervalUs);
+	double currentEdenHybridOverhead = calculateHybridEdenOverhead(env, _historicalPartialGCTime, currentCpuEdenOverhead);
+	double bestOverheadPrediction = currentEdenHybridOverhead;
+
+	/* How large the hops (in bytes) between samples should be */
+	uintptr_t samplingGranularity = (uintptr_t)(maxEdenChange - minEdenChange) / numberOfSamples;
+
+	/* Try "numberOfSamples" tests on the hybrid overhead curve, to determine which eden change will have best hybrid overhead */
+	for (uintptr_t i = 0; i < numberOfSamples; i++) {
+		/* Start from the right side of the curve */
+		intptr_t edenChange = (intptr_t)maxEdenChange - (samplingGranularity * i);
+
+		/* Predict what the pgc pause time, and gc overhead will be, if eden changes by 'edenChange' bytes*/
+		double estimatedCpuOverhead = predictCpuOverheadForEdenSize(env, currentIdealEdenSize, edenChange, freeTenure, avgPgcIntervalUs);
+		double estimatedPGCAvgTime = predictPgcTime(env, currentIdealEdenSize, edenChange);
+		double estimatedHybridOverhead = calculateHybridEdenOverhead(env, (uintptr_t)estimatedPGCAvgTime / 1000, estimatedCpuOverhead);
+		
+		if (estimatedHybridOverhead < bestOverheadPrediction) {
+			/* The hybrid between pgc pause time, and gc overhead (% time gc is active), is better than what was previously thought to be the best, save the eden size */
+			recommendedEdenChange = edenChange;
+			bestOverheadPrediction = estimatedHybridOverhead;
+		} 
+	}
+
+	uintptr_t recommendedSize = currentIdealEdenSize + recommendedEdenChange;
```

Comment:
Nitpick, but the method should be calculateRecommendedEdenChange... (not Size). You add currentIdealEdenSize here, and then at call site sub it.

Reply:
Agreed, it's a bit redundant to add it in one spot, and then subtract in the other.

Will rename the function, as well as make the function simply return the change in eden bytes, rather than suggested eden size (along with fixing up this callsite)